### PR TITLE
Update metaprogramming docs

### PR DESCRIPTION
**Pull request by nicolasstucki**
[skip test]
[skip test_windows_fast]
[skip community_build]

### CODE REVIEW

File: docs/docs/reference/metaprogramming/toc.md [hunk 1]
```diff
@@ -25,13 +25,13 @@ introduce the following fundamental facilities:
...
+3. [Runtime Staging](./staging.md) Where macros construct code at _compile-time_,
    staging lets programs construct new code at _runtime_. That way,
    code generation can depend not only on static data but also on data available at runtime. This splits the evaluation of the program in two or more phases or ...
    stages. Consequently, this method of generative programming is called "Multi-Stage Programming". Staging is built on the same foundations as macros. It uses
    quotes and splices, but leaves out `inline`.
 
-4. [TASTy Reflection](./tasty-reflect.md) Quotations are a "black-box"
+4. [Quote Reflection](./quote-reflection.md) Quotations are a "black-box"
    representation of code. They can be parameterized and composed using
    splices, but their structure cannot be analyzed from the outside. TASTy
```
Comment (b-studios) [inline docs/docs/reference/metaprogramming/toc.md:36]: Here you still use TASTy

**Review by b-studios — COMMENTED** [review]
What is your motivation of calling it "Quote reflection" and not just "reflection"? I find it a bit confusing since I understand quotation as a mechanism to turn programs into trees, but we do not reflect on the mechanism, but ob the trees themselves.

**Comment by nicolasstucki** [comment]
I added the `Quote` in `Quote Reflection` to distinguish it from `scala.reflect` and hint at the fact that the reflection is inside the `Quotes` (i.e. `quotes.reflect`). I'm fine with `Reflection` if that is clearer to others.

**Comment by b-studios** [comment]
Ok, thanks for the explanation. So the point is runtime reflection vs. compiletime reflection, right?

**Comment by b-studios** [comment]
Renaming might also conflict with PR #10932, so maybe we want to merge that one first?

**Comment by nicolasstucki** [comment]
Merged and rebased over #10932. Also fixed the Accumulator issue noticed in #10932.

**Comment by b-studios** [comment]
> I'm fine with Reflection if that is clearer to others.

Just to get another opinition, @liufengyun what do you think?

**Comment by liufengyun** [comment]
> > I'm fine with Reflection if that is clearer to others.
> 
> Just to get another opinion, @liufengyun what do you think?

Reflection seems to be simpler and clearer.

**Review by liufengyun — COMMENTED** [review]
While `Quote Reflection` can work as well, I find the single word `Reflection` is already clear enough, adding `Quote` does not seem to add more useful meaning to it. Maybe for simplicity just use `Reflection`.

**Comment by nicolasstucki** [comment]
Updated `Quote Reflection` to `Reflection`

**Review by b-studios — APPROVED** [review]

**Pull request merged**